File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/util/AccountDaoManager.java
/*
 * © 2016-2018 FlowCrypt Limited. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.util;

import com.flowcrypt.email.database.dao.source.AccountDao;

/**
 * @author Denis Bondarenko
 *         Date: 15.01.2018
 *         Time: 16:34
 *         E-mail: DenBond7@gmail.com
 */
public class AccountDaoManager {
    public static final String USER_WITH_MORE_THAN_21_LETTERS_ACCOUNT = "user_with_more_than_21_letters_account.json";

    public static AccountDao getDefaultAccountDao() {
        return TestGeneralUtil.readObjectFromResources("default_account.json", AccountDao.class);
    }

    public static AccountDao getAccountDao(String accountPath) {
        return TestGeneralUtil.readObjectFromResources(accountPath, AccountDao.class);
    }

    public static AccountDao getUserWitMoreThan21Letters() {
        return TestGeneralUtil.readObjectFromResources(USER_WITH_MORE_THAN_21_LETTERS_ACCOUNT, AccountDao.class);
    }
}

File: FlowCrypt/src/androidTest/java/com/flowcrypt/email/util/AccountDaoManager.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.util

import com.flowcrypt.email.api.retrofit.response.model.ClientConfiguration
import com.flowcrypt.email.database.entity.AccountEntity

/**
 * @author Denys Bondarenko
 */
class AccountDaoManager {
  companion object {

    fun getDefaultAccountDao(): AccountEntity {
      return getAccountDao("base_account_settings.json")
    }

    fun getAccountDao(accountPath: String): AccountEntity {
      return TestGeneralUtil.readObjectFromResources(accountPath, AccountEntity::class.java)
    }

    fun getUserWithMoreThan21Letters(): AccountEntity {
      return getUserFromBaseSettings("user_with_more_than_21_letters@flowcrypt.test")
    }

    fun getUserWithoutLetters(): AccountEntity {
      return getUserFromBaseSettings("user_without_letters@flowcrypt.test")
    }

    fun getUserWithSingleBackup(): AccountEntity {
      return getUserFromBaseSettings("single_backup@flowcrypt.test")
    }

    fun getUserWithoutBackup(): AccountEntity {
      return getUserFromBaseSettings("no_backups@flowcrypt.test")
    }

    fun getUserWithClientConfiguration(clientConfiguration: ClientConfiguration): AccountEntity {
      return getUserFromBaseSettings("user_with_client_configuration@flowcrypt.test")
        .copy(clientConfiguration = clientConfiguration)
    }

    fun getUserFromBaseSettings(user: String): AccountEntity {
      return getDefaultAccountDao().copy(
        email = user,
        smtpUsername = user,
        username = user,
      )
    }
  }
}
